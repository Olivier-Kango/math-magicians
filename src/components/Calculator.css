.Calculator ul li {
  list-style: none;
}
