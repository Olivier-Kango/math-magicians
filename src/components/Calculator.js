import React, { useState } from 'react';
import calculate from '../logic/calculate';

const h3 = {
  margin: '40px 0',
};

const container = {
  display: 'flex',
  justifyContent: 'space-between',
};

const Calculator = () => {
  const stateElement = {
    total: 0,
    next: null,
    operation: null,
  };

  const [items, setitems] = useState(stateElement);

  const handleClick = (e) => {
    const elt = e.target.innerText;
    const newitems = { ...items };
    setitems(calculate(newitems, elt));
  };

  return (
    <div style={container} className="container-calculator">
      <h3 style={h3}>Let do some Maths!</h3>
      <div className="container">
        <div className="result">{items.next || items.operation || items.total || 0}</div>
        <button type="button" className="left" onClick={handleClick}>AC</button>
        <button type="button" className="left" onClick={handleClick}>+/-</button>
        <button type="button" className="left" onClick={handleClick}>%</button>
        <button type="button" className="right" onClick={handleClick}>รท</button>
        <button type="button" className="left" onClick={handleClick}>7</button>
        <button type="button" className="left" onClick={handleClick}>8</button>
        <button type="button" className="left" onClick={handleClick}>9</button>
        <button type="button" className="right" onClick={handleClick}>x</button>
        <button type="button" className="left" onClick={handleClick}>4</button>
        <button type="button" className="left" onClick={handleClick}>5</button>
        <button type="button" className="left" onClick={handleClick}>6</button>
        <button type="button" className="right" onClick={handleClick}>-</button>
        <button type="button" className="left" onClick={handleClick}>1</button>
        <button type="button" className="left" onClick={handleClick}>2</button>
        <button type="button" className="left" onClick={handleClick}>3</button>
        <button type="button" className="right" onClick={handleClick}>+</button>
        <button type="button" className="zero" onClick={handleClick}>0</button>
        <button type="button" className="left" onClick={handleClick}>.</button>
        <button type="button" className="right" onClick={handleClick}>=</button>
      </div>
    </div>
  );
};

export default Calculator;
