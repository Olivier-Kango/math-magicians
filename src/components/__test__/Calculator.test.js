import { render, screen } from '@testing-library/react';
import React from 'react';
import Calculator from '../Calculator';

describe('<Calculator />', () => {
  it('shows numbers', () => {
    render(<Calculator />);
       
    const buttons = document.getElementsByClassName('left');

    Array.from(buttons).forEach((n) => {
      expect(screen.getByText(n.innerHTML.toString())).toBeInTheDocument();
    });
  });
});
