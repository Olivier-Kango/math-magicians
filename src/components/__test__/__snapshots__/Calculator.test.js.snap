// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calculator component shows numbers 1`] = `
<button
  class="left"
  type="button"
>
  AC
</button>
`;

exports[`Testing Calculator component shows numbers 2`] = `
<button
  class="left"
  type="button"
>
  +/-
</button>
`;

exports[`Testing Calculator component shows numbers 3`] = `
<button
  class="left"
  type="button"
>
  %
</button>
`;

exports[`Testing Calculator component shows numbers 4`] = `
<button
  class="left"
  type="button"
>
  7
</button>
`;

exports[`Testing Calculator component shows numbers 5`] = `
<button
  class="left"
  type="button"
>
  8
</button>
`;

exports[`Testing Calculator component shows numbers 6`] = `
<button
  class="left"
  type="button"
>
  9
</button>
`;

exports[`Testing Calculator component shows numbers 7`] = `
<button
  class="left"
  type="button"
>
  4
</button>
`;

exports[`Testing Calculator component shows numbers 8`] = `
<button
  class="left"
  type="button"
>
  5
</button>
`;

exports[`Testing Calculator component shows numbers 9`] = `
<button
  class="left"
  type="button"
>
  6
</button>
`;

exports[`Testing Calculator component shows numbers 10`] = `
<button
  class="left"
  type="button"
>
  1
</button>
`;

exports[`Testing Calculator component shows numbers 11`] = `
<button
  class="left"
  type="button"
>
  2
</button>
`;

exports[`Testing Calculator component shows numbers 12`] = `
<button
  class="left"
  type="button"
>
  3
</button>
`;

exports[`Testing Calculator component shows numbers 13`] = `
<button
  class="left"
  type="button"
>
  .
</button>
`;

exports[`Testing Calculator component shows operators 1`] = `
<button
  class="right"
  type="button"
>
  รท
</button>
`;

exports[`Testing Calculator component shows operators 2`] = `
<button
  class="right"
  type="button"
>
  x
</button>
`;

exports[`Testing Calculator component shows operators 3`] = `
<button
  class="right"
  type="button"
>
  -
</button>
`;

exports[`Testing Calculator component shows operators 4`] = `
<button
  class="right"
  type="button"
>
  +
</button>
`;

exports[`Testing Calculator component shows operators 5`] = `
<button
  class="right"
  type="button"
>
  =
</button>
`;
